### PR TITLE
Added semi-integration test

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Reasoning behind the pull request
- added semi-integration test
- added une function able to export an account's code & data
  
## Proposed changes
- added semi-integration test
- added une function able to export an account's code & data

## Testing procedure
- N/A

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/ElrondNetwork/elrond-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by schimih — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

File: debug/process/stateExport.go [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go/common"
+	"github.com/ElrondNetwork/elrond-go/state"
+)
+
+// ExportUserAccountState will try to export the account state of a provided address
+func ExportUserAccountState(accountsDB state.AccountsAdapter, identifier string, address []byte, parentDirForFiles string) {
```
Comment (bogdan-rosianu) [inline debug/process/stateExport.go:15]: might return error? otherwise, remove the `try` from the comment

File: debug/process/stateExport.go [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		log.Error("ExportUserAccountState", "identifier", identifier, "error", err)
+		return
+	}
+}
+
+func getCodeAndData(accountsDB state.AccountsAdapter, address []byte) (code []byte, csvHexedData []string, err error) {
+	account, err := accountsDB.GetExistingAccount(address)
+	if err != nil {
+		return nil, nil, fmt.Errorf("%w while getting existing data for hex address %s",
+			err, hex.EncodeToString(code))
```
Comment (bogdan-rosianu) [inline debug/process/stateExport.go:39]: wrong print. `address` instead of `code`

**Review by schimih — DISMISSED** [review]

File: debug/process/stateExport.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+		Directory:     parentDirForFiles,
+		Prefix:        computePrefix("data", identifier),
+		FileExtension: "hex",
+	}
+	f, err := core.CreateFile(fileArgs)
+	if err != nil {
+		return err
+	}
+
+	for _, line := range lines {
```
Comment (bogdan-rosianu) [inline debug/process/stateExport.go:148]: refactor. you should write only once to the file. see example:

```
        buffer := bytes.Buffer{}
	for _, line := range lines {
		buffer.Write([]byte(fmt.Sprintf("%s\n", line)))
	}

	_, _ = f.Write(buffer.Bytes())
```
I've done some tests. your code is ~10 times slower

File: integrationTests/vm/txsFee/scCalls_test.go [hunk 4]
```diff
@@ -300,3 +362,47 @@ func TestESDTScCallAndGasChangeShouldWork(t *testing.T) {
...
+	endTime := time.Now()
+	assert.Nil(t, errProcess)
+	assert.Equal(t, vmcommon.Ok, returnCode)
+
+	_, errCommit := testContext.Accounts.Commit()
+	require.Nil(t, errCommit)
+
+	intermediateTxs := testContext.GetIntermediateTransactions(t)
+	assert.Equal(t, 1, len(intermediateTxs))
+	log.Info(spew.Sdump(intermediateTxs))
```
Comment (bogdan-rosianu) [inline integrationTests/vm/txsFee/scCalls_test.go:402]: might print them in a prettier way (bech32 addresses, and so on)

File: integrationTests/vm/txsFee/utils/utils.go [hunk 5]
```diff
@@ -299,3 +343,51 @@ func GenerateUserNameForMyDNSContract() []byte {
...
+// ApplyDataOverwritingExistingData applies pairs of <key,value> from provided file to the state of the provided address
+// Before applying the data it does a cleanup on the old state
+// the data from the file must be in the following format:
+//hex(key1),hex(value1)
+//hex(key2),hex(value2)
+//...
+// Example:
+//61750100,0000
+//61750101,0001
+func ApplyDataOverwritingExistingData(tb testing.TB, testContext *vm.VMTestContext, address []byte, pathToData string) {
```
Comment (bogdan-rosianu) [inline integrationTests/vm/txsFee/utils/utils.go:356]: a better name would be `OverwriteAccountStorageWithHexFileContent`

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: debug/process/stateExport.go (hunk 1 above)
Comment (iulianpascalau) [reply debug/process/stateExport.go:15]: no, it is meant to easily usable when modifying the code ad-hoc

File: debug/process/stateExport.go (hunk 2 above)
Comment (iulianpascalau) [reply debug/process/stateExport.go:39]: changed

File: debug/process/stateExport.go (hunk 3 above)
Comment (iulianpascalau) [reply debug/process/stateExport.go:148]: good suggestion, applied

File: integrationTests/vm/txsFee/scCalls_test.go (hunk 4 above)
Comment (iulianpascalau) [reply integrationTests/vm/txsFee/scCalls_test.go:402]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by schimih — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**